### PR TITLE
Filter functionality with category

### PR DESCRIPTION
**Pull request by dev-enforced**
In this Pull Request one can find the code for filters functionality provided in the Listed Products of Eleganz Stores Web Application. And the feature of home page categories has been added where one can click on any of the categories and get redirected to the  product listing page with the category they clicked on the home page.

Live preview link : https://deploy-preview-7--eleganz-store.netlify.app/products
For checking the category functionality: https://deploy-preview-7--eleganz-store.netlify.app/ In the categories section click on any of the category

### CODE REVIEW

File: src/services/Products/getProductsAndCategories.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+const loadProductsAndCategories = async () => {
+    try {
+        const { data: productsData } = await axios.get("/api/products");
+        const { data: categoriesData } = await axios.get("/api/categories");
+        return [productsData,categoriesData]
+    } catch {
+        console.log("API Call for Product and Category Error");
+    }
```
Comment (mariyasada) [inline src/services/Products/getProductsAndCategories.js]: liked that you have created one function for fetching product and categories

**Review by mariyasada — APPROVED** [review]
good to go

File: src/services/Products/getProductsAndCategories.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+const loadProductsAndCategories = async () => {
+    try {
+        const { data: productsData } = await axios.get("/api/products");
+        const { data: categoriesData } = await axios.get("/api/categories");
+        return [productsData,categoriesData]
```
Comment (tushartiwari7) [inline src/services/Products/getProductsAndCategories.js]: Suggstion: You can use `Promise.all` here.

File: src/utilities/filterUtilities.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+const applySortingChoice = (givenState, providedProducts) => {
+    if (givenState.sortPriceOrder === "Low To High") {
+        return [...providedProducts].sort((a, b) => Number(a.discountedPrice) - Number(b.discountedPrice))
+    } else if (givenState.sortPriceOrder === "High To Low") {
+        return [...providedProducts].sort((a, b) => Number(b.discountedPrice) - Number(a.discountedPrice))
+    } else {
+        return [...providedProducts]
```
Comment (tushartiwari7) [inline src/utilities/filterUtilities.js:7]: suggestion: While returning the exact same arr you don't need to create a copy first.

**Review by tushartiwari7 — APPROVED** [review]
Code Quality is good, liked the impementation of utility fuctions.
Just suggested some points, have a look.

**Pull request merged**